_posts/2022/2022-11-14-safely-remove-a-column-field-from-active-record.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Safely Remove a Column from an Active Record Model"
+description: "Avoid migration-related errors when you’re tidying your database"
+layout: article
+category: ruby
+image:
+  base: "2022/safely-remove-a-column-field-from-active-record"
+  alt: "A partial column against flat background"
+  credit: "ANIRUDH"
+  source: "https://unsplash.com/photos/jtiQCAAzLYY"
+---
+
+Adding and deploying additional columns to an Active Record model is easy. Typically the deployment runs your migrations and then the new code that uses that database column is released afterward.
+
+However, when removing a column you can often have errors as the old code (that still references the column) will be running as your migration removes the column from the database. This is because Active Record caches database columns when it spins up a Rails application, without the column actually on the database table it causes exceptions until your app reboots or redeploys.
+
+To avoid this, it's worth a multi-step strategy when removing columns in your database.
+
+
+## Use...
+
+...a multi-step strategy.
+
+First tell Active Record to ignore the column.
```

Comment:
We're assuming the code is no longer using the old column anywhere, right?

Reply:
```suggestion
First tell Active Record to ignore the column. This stops the rest of you application being able to reference this column.
```